### PR TITLE
C++11 patchset 2: remove util/cpp11.h and util/cpp11_container.h

### PR DESCRIPTION
**Pull request by nerzhul**
Requires #5814 to be merged, to review just look at third commit

some long types on unordered_set/map have now typedefs to reduce coding lines

### CODE REVIEW

**Comment by nerzhul** [comment]
I will rebase this when the other PR will be merged or if i need to do some modifications

**Comment by nerzhul** [comment]
rebased

File: src/face_position_cache.cpp [hunk 1]
```diff
@@ -21,14 +21,14 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "threading/mutex_auto_lock.h"
 
 
-UNORDERED_MAP<u16, std::vector<v3s16> > FacePositionCache::cache;
+std::unordered_map<u16, std::vector<v3s16> > FacePositionCache::cache;
```
Comment (SmallJoker) [inline src/face_position_cache.cpp]: Either
`namespace::container<typename, namespace::container<typename>>` or
`namespace::container<typename, namespace::container<typename> >`

But you mixed both notations in this PR.

**Comment by nerzhul** [comment]
Exact, well done, i will fix it this evening. On c++11 the space isn't required

**Pull request merged**